spec/lib/pub_sub/message_spec.rb

```diff
@@ -64,27 +64,29 @@ class EntityUpdate
     class EntityUpdate
     end
 
-    context 'on unknown subscriptions' do
+    context 'quietly' do
       let(:payload) { raw_sns_message['body'] }
-      it 'quietly' do
+
+      it 'on unknown senders' do
         allow(PubSub.config.subscriptions).to receive(:[]).with(
           'entity-service-prod'
         ).and_return(nil)
-        allow(PubSub.config).to receive_message_chain("logger.warn").and_return(anything())
+        allow(PubSub.config).to receive_message_chain("logger.error").and_return(anything())
         # first, make sure the validator throws an exception
         expect{subject.validate_message!}.to raise_error(PubSub::ServiceUnknown)
         # then, make sure `process` does not
         subject.process
       end
-    end
 
-    context 'on unknown message types' do
-      let(:payload) { raw_sns_message['body'] }
-      it 'with exception' do
+      it 'on unknown types' do
         allow(PubSub.config.subscriptions).to receive(:[]).with(
           'entity-service-prod'
         ).and_return(['unknown_type'])
-        expect{subject.process}.to raise_error(PubSub::MessageTypeUnknown)
+        allow(PubSub.config).to receive_message_chain("logger.error").and_return(anything())
+        # first, make sure the validator throws an exception
+        expect{subject.validate_message!}.to raise_error(PubSub::MessageTypeUnknown)
```

Comment:
At the moment, all this logic is inter-connected, so testing one without the other is of little use. I'll keep things as is for now but will make a mental note to possibly revisit later